full-service/src/service/wallet_trait.rs

```diff
@@ -0,0 +1,276 @@
+use crate::service::decorated_types::{JsonWalletStatus, JsonCreateAccountResponse, JsonAccount, JsonTxo, JsonBalanceResponse, JsonAddress, JsonSubmitResponse, JsonTransactionLog, JsonBlock, JsonBlockContents, JsonProof};
+use crate::service::WalletServiceError;
+use crate::db::account::AccountID;
+use mc_mobilecoind_json::data_types::{JsonTxProposal, JsonTx, JsonTxOut};
+
+#[cfg(test)]
+use mockall::*;
+
+
+/// A MobileCoin wallet.
+#[cfg_attr(test, automock)]
+pub trait Wallet {
```

Comment:
This is new.

I'm working on filling in the ??? placeholders in the doc comments, but suggested text is welcome!

Reply:
Feel free to look at the [README](https://github.com/mobilecoinofficial/full-service/blob/main/README.md) - many of the ??? fields below are specified there.